### ISSUE TITLE
Type of "send" is partially unknown

### ISSUE THREAD

**Issue by rijenkii** [cole/aiosmtplib#267]
**Describe the bug**
Calling `pyright` in strict mode on `aiosmtplib.send` displays a following error:
`
Type of "send" is "(message: EmailMessage | Message[str, str] | str | bytes, /, *, sender: str | None = None, recipients: str | Sequence[str] | None = None, mail_options: Sequence[str] | None = None, rcpt_options: Sequence[str] | None = None, hostname: str | None = "localhost", port: int | None = None, username: str | bytes | None = None, password: str | bytes | None = None, local_hostname: str | None = None, source_address: Tuple[str, int] | None = None, timeout: float | None = DEFAULT_TIMEOUT, use_tls: bool = False, start_tls: bool | None = None, validate_certs: bool = True, client_cert: str | None = None, client_key: str | None = None, tls_context: SSLContext | None = None, cert_bundle: str | None = None, socket_path: str | bytes | PathLike[Unknown] | None = None, sock: socket | None = None) -> Coroutine[Any, Any, Tuple[Dict[str, SMTPResponse], str]]" (reportUnknownMemberType)
`

Relevant part is `socket_path: str | bytes | PathLike[Unknown] | None = None`.

**To Reproduce**
```
$ cat main.py 
# pyright: strict

import aiosmtplib

async def main():
    await aiosmtplib.send("test")

$ pyright main.py
/tmp/smtp/main.py
  /tmp/smtp/main.py:6:11 - error: Type of "send" is partially unknown
    Type of "send" is "(message: EmailMessage | Message[str, str] | str | bytes, /, *, sender: str | None = None, recipients: str | Sequence[str] | None = None, mail_options: Sequence[str] | None = None, rcpt_options: Sequence[str] | None = None, hostname: str | None = "localhost", port: int | None = None, username: str | bytes | None = None, password: str | bytes | None = None, local_hostname: str | None = None, source_address: Tuple[str, int] | None = None, timeout: float | None = DEFAULT_TIMEOUT, use_tls: bool = False, start_tls: bool | None = None, validate_certs: bool = True, client_cert: str | None = None, client_key: str | None = None, tls_context: SSLContext | None = None, cert_bundle: str | None = None, socket_path: str | bytes | PathLike[Unknown] | None = None, sock: socket | None = None) -> Coroutine[Any, Any, Tuple[Dict[str, SMTPResponse], str]]" (reportUnknownMemberType)
1 error, 0 warnings, 0 informations 
```

**Comment by cole** [comment]
Thanks for reporting! I'm not sure how to fix this while still supporting 3.8. If I switch to `Union[str, bytes, os.PathLike[Union[str, bytes]]]`, it breaks mypy.

If I do something like:

```
if sys.version_info[:2] < (3, 9):
    SocketPathType = Union[str, bytes, os.PathLike]
else:
    SocketPathType = Union[str, bytes, os.PathLike[Union[str, bytes]]]
```

mypy is happy, but pyright isn't.

**Comment by rijenkii** [comment]
Try removing `[:2]` after `sys.version_info`,

**Comment by cole** [comment]
ah, thanks, I missed that. Should be fixed in 3.0.2
